### PR TITLE
040 patch: add logout

### PR DESCRIPTION
**Pull request by ntotten-kainos**
Making required changes to frontend code to enable log out.

- [x] Unit tests passing locally
- [x] Build passing locally
- [x] Build + Tests passing on remote

Further locking-down of users from pages other than the Login page will be handled by the Authorisation ticket.

### CODE REVIEW

File: src/controllers/AuthController.ts [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
     } catch (error) {
         res.locals.errormessage = error.message;
+        res.locals.loggedin = false;
         res.render('loginForm', req.body);
     }
+}
+
+export const logout = async(req: express.Request, res: express.Response): Promise<void> => {
+    // Wipe out the token from session storage.
+    console.log(req.session.token);
```
Comment (boydstepkainos) [inline src/controllers/AuthController.ts:23]: needs removed (all console.logs please)

Comment (ntotten-kainos) [reply src/controllers/AuthController.ts:23]: Ah yep! Rookie error lol

**Review by mark-mcilroykainos — APPROVED** [review]

**Review by ntotten-kainos — COMMENTED** [review]
Ready for review

**Review by mark-mcilroykainos — DISMISSED** [review]

**Review by mark-mcilroykainos — APPROVED** [review]

**Review by boydstepkainos — DISMISSED** [review]

**Comment by ntotten-kainos** [comment]
Had to fix merge conflicts created by JobRoleController and corresponding test class

**Comment by ntotten-kainos** [comment]
Another wee test fix

**Comment by ntotten-kainos** [comment]
Fixing more merge conflicts

**Review by mark-mcilroykainos — APPROVED** [review]

File: src/controllers/JobRoleController.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { getJobRoles } from "../services/JobRoleService";
 
 export const getAllJobRoles = async (req: express.Request, res: express.Response): Promise<void> => {
     try {
+        res.locals.loggedin = true;
```
Comment (shaunganley) [inline src/controllers/JobRoleController.ts:6]: you can avoid having to set this new var everywhere by passing req.session.token into the view and checking if it set or not

Comment (ntotten-kainos) [reply src/controllers/JobRoleController.ts]: Added some new middleware that I think does this :) it's doing a boolean representation of req.session.token through !!req.session.token and setting this as the res.locals.loggedin value as part of every request made.

**Review by mark-mcilroykainos — APPROVED** [review]
Looks good Nathan, all tested locally and works as expected

**Review by boydstepkainos — APPROVED** [review]

**Pull request merged**